File: Starters 66 Division 3 (Rated)/c.cpp
#include <iostream>
using namespace std;

int main()
{
    int t;
    cin >> t;
    while (t--)
    {
        int n, m, k, x,val;
        cin >> n >> m >> k >> x;
        val = x % (n * k + m);
        if (val != 0 && (val - (n * (k - 1))) <= 0)
        {
            cout << "NO" << endl;
        }
        else
            cout << "YES" << endl;
    }
    return 0;
}

File: Starters 66 Division 3 (Rated)/c.py
for _ in range(int(input())):
	n, m, k, x = map(int, input().split())
	v = x % (n*k + m)
	if v != 0 and (v - n*(k-1)) <= 0:
		print("NO")
	else:
		print("YES")
